src/components/core/posts/SinglePost.tsx

```diff
@@ -102,6 +104,26 @@ export default function SinglePost({
     }
   };
 
+  const handleComment = () => {
+    setShowCommentBox(!showCommentBox);
+    console.log("comment");
+  };
+
+  const uploadComment = async (id: string,comment_message: string) => {
+    console.log("pastcomments" + singlePost.comments);
+    const Comments = [...singlePost.comments, comment_message];
+    console.log("present comments" + singlePost.comments);
+    const comment = await addComment(id , Comments);
+    console.log(comment);
+  };
+
+  const [comment_message, setComment_message] = useState("");
```

Comment:
useState should be declared above the functions as this will give error in production

Reply:
Done

Reply:
I have placed useState variables above functions